test/mock-interceptor-unused-assertions.js

```diff
@@ -0,0 +1,138 @@
+'use strict'
```

Comment:
there are no tests for `persist`, can you add?

Also, I think a persisted interceptor should error if it's _never_ called (I don't know if the logic does that today)

Reply:
> there are no tests for persist, can you add?

Yep!

> Also, I think a persisted interceptor should error if it's never called (I don't know if the logic does that today)

I don't think that's currently possible with the way Undici counts requests, since the `times` value is only set if `times(<number>)` has been called. In this PR I "cheat" by treating `times` as `1` if neither `persist` nor `times` is set.

It could definitely be solved by tracking the call count counting upwards to either a `limit` set by `times()`, `1` if neither `persist` nor `times` is set, or forever if `persist` is set, before it stops serving the mocked response.

I can give it a try, but that feels like it might belong in a separate pull request. On the other hand, it would trip on a new assertion and thus be a breaking change to the API introduced in this PR, so maybe it belongs here…

Just so I don't waste time on code that will be rejected: @mcollina or @ronag are you open to me changing the call count tracking in this PR so it can all land at once?

Reply:
Go for it!

Reply:
Done 😅